pkg/controllers/osc/resources/secrets.go

```diff
@@ -19,22 +19,28 @@ package resources
 import (
 	"fmt"
 
+	"k8c.io/operating-system-manager/pkg/generator"
 	"k8c.io/operating-system-manager/pkg/resources/reconciling"
 
 	corev1 "k8s.io/api/core/v1"
 )
 
-// CloudInitSecretCreator returns a function to create a secret that contains the cloud-init configurations.
-func CloudInitSecretCreator(mdName string, oscType CloudInitSecret, data []byte) reconciling.NamedSecretCreatorGetter {
+// CloudConfigSecretCreator returns a function to create a secret that contains the cloud-init or ignition configurations.
+func CloudConfigSecretCreator(mdName string, osName string, oscType CloudConfigSecret, data []byte) reconciling.NamedSecretCreatorGetter {
 	return func() (string, reconciling.SecretCreator) {
 		secretName := fmt.Sprintf(MachineDeploymentSubresourceNamePattern, mdName, oscType)
 		return secretName, func(sec *corev1.Secret) (*corev1.Secret, error) {
 			if sec.Data == nil {
 				sec.Data = map[string][]byte{}
 			}
 
-			sec.Data["cloud-init"] = data
-
+			switch generator.GetProvisioningUtility(osName) {
```

Comment:
Why knowing the cloud-config type in a secret is important? I mean what do we get by having two keys here? I would say lets keep cloud-init. Or maybe cloud-config.